xarray/core/indexing.py

```diff
@@ -840,10 +863,18 @@ def __init__(self, array):
     def _ensure_cached(self):
         self.array = as_indexable(self.array.get_duck_array())
 
+    async def _async_ensure_cached(self):
+        duck_array = await self.array.async_get_duck_array()
+        self.array = as_indexable(duck_array)
+
     def get_duck_array(self):
         self._ensure_cached()
```

Comment:
`_ensure_cached` seems like pointless indirection, it is only used once. let's consolidate.

Reply:
Removed in https://github.com/pydata/xarray/pull/10327/commits/884ce139acfe3c6b1f38ce1eb8ac03370fd4e9dd, but I still feel like it could be simplified further. Does it really need to have the side-effect of re-assigning to `self.array`?